org.lflang/src/org/lflang/validation/AttributeSpec.java

```diff
@@ -147,11 +149,7 @@ private Set<String> processNamedAttrParms(LFValidator validator, Attribute attr)
      * @param type The type of the parameter
      * @param defaultValue If non-null, parameter is optional.
      */
-    record AttrParamSpec(String name, AttrParamType type, Object defaultValue) {
-
-        private boolean isOptional() {
-            return defaultValue == null;
-        }
+    record AttrParamSpec(String name, AttrParamType type, boolean isOptional) {
```

Comment:
The default value was not used anywhere and since this class only works as a spec for the verifier I would argue that the default value is not needed anyway. It would only make sense to keep it if we use this spec in `AttributUtils' to retrieve the value of the parameter, but this would be a larger change.